axelor-account/src/main/resources/views/InvoiceLine.xml

```diff
@@ -190,6 +191,16 @@
 
 		<attribute name="readonly" for="accountingPanel" expr="eval: invoice.statusSelect &gt; 2"/>
 	</action-attrs>
+	
+	<action-attrs name="action-invoice-line-attrs-set-product-domain-onselect">
+		<attribute name="domain" for="product" expr="eval: isFilterOnSupplier ? &quot; self.isModel = false and self.id in (${__parent__?.partner.supplierCatalogList.product.collect{it.id}.join(',')}) &quot; : &quot; self.isModel = false &quot;" 
+			if="__parent__.operationTypeSelect &lt; 3  &amp;&amp; __config__.app.isApp('purchase')  &amp;&amp;  __config__.app.getApp('purchase').getManageSupplierCatalog()"/>
+		<attribute name="domain" for="product" expr="eval: &quot; self.isModel = false &quot;" if="__parent__.operationTypeSelect &gt; 2"/>
+	</action-attrs>
+	
+	<action-attrs name="action-invoice-line-attrs-hide-filter">
```

Comment:
Remove this action-attrs and put attribute on action-attrs-account-invoice-line-onnew-onload, as this action called on onNew and onLoad